vwifi.c

```diff
@@ -453,7 +459,36 @@ static int owl_get_station(struct wiphy *wiphy,
                            const u8 *mac,
                            struct station_info *sinfo)
 {
-    /* TODO: Get station infomation */
+    struct owl_context *owl = ndev_get_owl_context(dev)->owl;
+    struct owl_ndev_priv_context *np = NULL;
+
+    /* Find the station which is being searched */
+    np = list_first_entry(&owl->netintf_list, struct owl_ndev_priv_context,
+                          list);
+    if (!ether_addr_equal(np->ndev->dev_addr, mac))
+        np = list_last_entry(&owl->netintf_list, struct owl_ndev_priv_context,
+                             list);
+
+    sinfo->filled = BIT_ULL(NL80211_STA_INFO_TX_PACKETS) |
+                    BIT_ULL(NL80211_STA_INFO_RX_PACKETS) |
+                    BIT_ULL(NL80211_STA_INFO_TX_FAILED) |
+                    BIT_ULL(NL80211_STA_INFO_TX_BYTES) |
+                    BIT_ULL(NL80211_STA_INFO_RX_BYTES) |
+                    BIT_ULL(NL80211_STA_INFO_SIGNAL) |
+                    BIT_ULL(NL80211_STA_INFO_INACTIVE_TIME);
+    if (np->is_assoc) {
+        sinfo->filled |= BIT_ULL(NL80211_STA_INFO_CONNECTED_TIME);
+        sinfo->connected_time =
+            jiffies_to_msecs(jiffies - np->conn_time) / 1000;
+    }
+    sinfo->tx_packets = np->stats.tx_packets;
+    sinfo->rx_packets = np->stats.rx_packets;
+    sinfo->tx_failed = np->stats.tx_dropped;
+    sinfo->tx_bytes = np->stats.tx_bytes;
+    sinfo->rx_bytes = np->stats.rx_bytes;
+    /* For CFG80211_SIGNAL_TYPE_MBM, value is expressed in dbm */
+    sinfo->signal = rand_int(-100, -30);
```

Comment:
I am afraid of the jitters because of the plain use of random numbers.
Can you follow the way how existing simulations draw a random number from a uniform distribution? 
See https://www.nsnam.org/docs/models/html/wifi-design.html

Reply:
OK. I'll take some time to read it.

Reply:
OK. I just read the paper (http://www.iro.umontreal.ca/~lecuyer/myftp/papers/streams00.pdf) to figure out the method (called RNG) to generate number in uniform distribution used by ns-3 project. It's fairly complicated, so I want to implement both RNG and sine function and compare the result of both.

Reply:
You don't have to implement RNG used in ns-3. Instead, you can concentrate on the introduction of a *smooth* function which is differentiable everywhere (hence continuous). It is quite useful for existing tools to render RSSI.

Reply:
Ok, I'll concentrate on it.